mantle/platform/api/aws/images.go

```diff
@@ -329,21 +329,28 @@ func (a *API) CreateImportRole(bucket string) error {
 	return nil
 }
 
-func (a *API) CreateHVMImage(snapshotID string, diskSizeGiB uint, name string, description string, architecture string) (string, error) {
+func (a *API) CreateHVMImage(snapshotID string, diskSizeGiB uint, name string, description string, architecture string, volumetype string, imdsv2Only bool) (string, error) {
 	var awsArch string
+	var bootmode string
 	if architecture == "" {
 		architecture = runtime.GOARCH
 	}
 	switch architecture {
 	case "amd64", "x86_64":
 		awsArch = ec2.ArchitectureTypeX8664
+		bootmode = "uefi-preferred"
```

Comment:
I think this also may need to be parameterized so that we leave RHCOS the same for now?

Reply:
Yeah, I left out the bootmode bit just to make sure I was on the right track. I will add that shortly.

Reply:
> I think this also may need to be parameterized so that we leave RHCOS the same for now?

I notice this comment is now marked as outdated so maybe I'm looking at the wrong information, but just based on your comment I think we want the new defaults to go to latest development RHCOS don't we? We can back out the changes with image.yaml config if we find one of the new settings is not palatable for whatever reason.

Reply:
> > I think this also may need to be parameterized so that we leave RHCOS the same for now?
> 
> I notice this comment is now marked as outdated so maybe I'm looking at the wrong information, but just based on your comment I think we want the new defaults to go to latest development RHCOS don't we?

I'm not sure. The last on this was Colin's suggestion in https://github.com/coreos/coreos-assembler/pull/3402 to make it opt-in, but I'm not sure if I've missed some conversations about it since then to track that change in OCP.

Reply:
yeah. We probably do need some paperwork done in the OCP side. @mike-nguyen - could you help us track what we'd need to do here?

Either way I think I'd prefer we land the defaults the way it currently is and then opt-out in the development head in openshift/os, which isn't hard to do.

Reply:
My thoughts offhand:

- the gp3 change can probably just sail through  - IMO it's sufficient to tag it for release notes
- The IMDSv2 change seems not unlikely to actually break some things; until recently IIRC some in-tree code in kubelet actually talked to the cloud-init user data too (to e.g. determine which aws region it's in).  There are potentially other operators.  Some of that may be IMDSv2 ready.  Others may not and it may sadly need some mechanism to opt out (openshift-install config and/or cluster-api/machineset?)
- UEFI is probably OK going to just sail through too - IMO it's sufficient to tag it for release notes

Reply:
IOW of these 3 things, I think only IMDSv2 may be worthy of an OCP enhancement, but it can probably be a small/quick one.